File: backend/src/domain/entities/FundExpense.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('fund_expenses')
export class FundExpense {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'fund_id' })
  fundId: number;

  @Column({ type: 'varchar', nullable: true })
  title: string;

  @Column({ type: 'decimal', nullable: true, precision: 15, scale: 2 })
  amount: number;

  @Column({ type: 'date', name: 'expense_date', nullable: true })
  expenseDate: Date;

  @Column({ type: 'varchar', name: 'proof_document_url', nullable: true })
  proofDocumentUrl: string;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend/src/main/java/com/service/backend/domain/entity/FundExpense.java
package com.service.backend.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("fund_expenses")
public class FundExpense {

    @Id
    private Long id;

    @Column("fund_id")
    private Long fundId;

    private String title;

    private BigDecimal amount;

    @Column("expense_date")
    private LocalDate expenseDate;

    @Column("proof_document_url")
    private String proofDocumentUrl;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
